sphinx/util/fileutil.py

```diff
@@ -35,14 +51,13 @@ def copy_asset_file(source: str, destination: str,
         # Use source filename if destination points a directory
         destination = os.path.join(destination, os.path.basename(source))
 
-    if source.lower().endswith('_t') and context is not None:
+    if _template_basename(source) and context is not None:
```

Comment:
(given earlier suggestion to make ``_template_basename`` always return *something*)
```suggestion
    if _template_basename(source) != source and context is not None:
```

Reply:
This seemed to make sense to me earlier, but after looking again, there seem to be some subtleties around directories-vs-filenames as destination that the previous implementation was handling.

The fact that that wasn't obvious is not ideal, so a better fix would make the reasoning more transparent (either via comments, or preferably by restructuring the logic).

Reply:
I think that adding a docstring to the `_template_basename` method would be a decent approach to resolve this (many IDEs would display that when developers visit the callsites).

Reply:
(`_template_basename` method docstring added)